File: 09.Problems on Arrays/11 Find the missing numbers/brute.py
def missing_number(a, n):
    # outer loop that runs from 1 to n
    for i in range(1, n+1):
        # flag varible to check if an element exists
        flg = 0
        
        # search the element using linear search
        for j in range(len(a)):
            if a[j] == i:
                # i is present in the array
                flg = 1
                break
        
        # check if the element is missing or not
        if flg == 0:
            return i 
    
    # to avoid warning 
    return -1 # this won't be executed

def main():
    n = int(input())
    a = list(map(int, input().split()))
    
    ans = missing_number(a, n)
    print("missing number: ", ans)
    
if __name__ == '__main__':
    main()
    
# Time complexity : O(n^2)

File: 09.Problems on Arrays/11 Find the missing numbers/brute.cpp
#include<bits/stdc++.h>
using namespace std;

int missingNumebr(int n, vector<int>&v){
    for(int i = 1; i <= n; ++i){
        int flg = 0;
        for(int j = 0; j < n-1; j++){
            if(v[j] == i){
                flg = 1;
                break;
            }
        }
        if(flg == 0) return i;
    }
    return -1;
}

int main(){
    int n; cin >> n;
    vector<int>v(n-1);
    for(int i = 0; i < n-1; ++i) cin >> v[i];

    int ans = missingNumebr(n, v);
    cout << "missing numebr: " << ans << '\n';

    return 0;
}

//  Time Complexity: O(N^2) where N = size of the array + 1
